public/app/core/components/FolderFilter/FolderFilter.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { css } from '@emotion/css';
+import { debounce } from 'lodash';
+import { AsyncMultiSelect, Icon, resetSelectStyles, useStyles2 } from '@grafana/ui';
+import { GrafanaThemeV2, SelectableValue } from '@grafana/data';
+
+import { FolderInfo } from 'app/types';
+import { getBackendSrv } from 'app/core/services/backend_srv';
+
+export interface FolderFilterProps {
+  onChange: (folder: FolderInfo[]) => void;
+  maxMenuHeight?: number;
+}
+
+export function FolderFilter({ onChange: propsOnChange, maxMenuHeight }: FolderFilterProps): JSX.Element {
+  const styles = useStyles2(getStyles);
+  const [loading, setLoading] = useState(false);
+  const loadOptions = useCallback((searchString: string) => loadOptionsAsync(searchString, setLoading), []);
+  const debouncedLoadOptions = useMemo(() => debounce(loadOptions, 300, { leading: true, trailing: true }), [
```

Comment:
Not sure if it's relevant here, but it might be good idea to use `debounced-promise` here instead. We had an issue with debouncing load function in Select before: https://github.com/grafana/grafana/pull/30706.

Reply:
Great catch will look into it.

Reply:
Changed to `debounced-promise` instead, as suggested.